.github/workflows/build-all-matrix.yaml

```diff
@@ -0,0 +1,188 @@
+
+      - name: Docker Login to GitHub Container Registry
+        if: ${{ env.REGISTRY == 'ghcr.io' }}
+        uses: docker/login-action@v3
+        with: { registry: "ghcr.io", username: "${{ github.repository_owner }}", password: "${{ secrets.GITHUB_TOKEN }}" }
+
+      - name: Build and Push LinuxKit containers for ${{matrix.docker_arch}}
+        env:
+          DOCKER_ARCH: "${{ matrix.docker_arch }}"
+          DO_PUSH: "yes"
+        run: bash build.sh linuxkit-containers
+  
+  build-kernels:
+    needs: [ matrix_prep ] # depend on the previous job...
+    runs-on: "${{ matrix.runner }}" # the runner to use is determined by the 'gha-matrix' code
+    strategy:
+      fail-fast: false # let other jobs try to complete if one fails, kernels might take long, and they'd be skipped on the next run
+      matrix:
+        include: ${{ fromJSON(needs.matrix_prep.outputs.kernels_json) }}
+    name: "Kernel ${{ matrix.kernel }}"
+    steps:
+      - name: Checkout build repo
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx # nb: no need for qemu here, kernels are cross-compiled, instead of the compilation being emulated
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker Login to quay.io
+        if: ${{ env.REGISTRY == 'quay.io' }}
+        uses: docker/login-action@v3
+        with: { registry: "quay.io", username: "${{ secrets.QUAY_USERNAME }}", password: "${{ secrets.QUAY_PASSWORD }}" }
+
+      - name: Docker Login to GitHub Container Registry
+        if: ${{ env.REGISTRY == 'ghcr.io' }}
+        uses: docker/login-action@v3
+        with: { registry: "ghcr.io", username: "${{ github.repository_owner }}", password: "${{ secrets.GITHUB_TOKEN }}" }
+
+      - name: Build and push Kernel ${{matrix.kernel}} (${{ matrix.arch }})
+        env:
+          DO_PUSH: "yes"
+        run: bash build.sh build-kernel "${{ matrix.kernel }}"
+
+  build-hook-ensemble:
+    needs: [ matrix_prep, build-linuxkit-containers, build-kernels ] # depend on the previous job...
+    runs-on: "${{ matrix.runner }}" # the runner to use is determined by the 'gha-matrix' code
+    strategy:
+      fail-fast: false # let other jobs try to complete if one fails
+      matrix:
+        include: ${{ fromJSON(needs.matrix_prep.outputs.lk_hooks_json) }}
+    name: "Hook ${{ matrix.kernel }}"
+    steps:
+      - name: Checkout build repo
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx # nb: no need for qemu here, kernels are cross-compiled, instead of the compilation being emulated
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker Login to DockerHub # read-only token, required to be able to pull all the linuxkit pkgs without getting rate limited.
+        if: ${{ env.LOGIN_TO_DOCKERHUB == 'yes' }}
+        uses: docker/login-action@v3
+        with: { registry: "docker.io", username: "${{ secrets.DOCKER_USERNAME }}", password: "${{ secrets.DOCKER_PASSWORD }}" }
```

Comment:
FYI, I've added dockerhub read only access creds for a Tinkerbell org account. `DOCKERHUB_USERNAME ` and `DOCKERHUB_PASSWORD `

Reply:
Nice!

Reply:
Set the default to use them and removed the conditional.